pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -180,7 +184,16 @@ func (p *staticPolicy) AddContainer(s state.State, pod *v1.Pod, container *v1.Co
 			return nil
 		}
 
-		cpuset, err := p.allocateCPUs(s, numCPUs)
+		//Call Topology Manager to get Container affinity
+		containerTopologyHint := p.affinity.GetAffinity(string(pod.UID), container.Name)
+		klog.Infof("[cpumanager] Pod %v, Container %v Topology Affinity is: %v", pod.UID, container.Name, containerTopologyHint)
+
+		sockets := make(map[int]bool)
+		socketsArray := containerTopologyHint.SocketMask.GetSockets()
+		for _, socket := range socketsArray {
+			sockets[socket] = true
+		}
+		cpuset, err := p.allocateCPUs(s, numCPUs, sockets)
```

Comment:
This seems like a strange indirection here. Why not just change `allocateCPUs()` to take a socketMask?

Reply:
Done

Reply:
It doesn't seem like this was changed.

Reply:
Overwrote it again by mistake, should be ok now